### PR TITLE
Add support for CGDisplayMode & safe CGDisplay wrapper

### PR DESCRIPTION
**Pull request by msanders**
This change is [Reviewable](https://reviewable.io/reviews/servo/core-graphics-rs/97)

### CODE REVIEW

File: src/display.rs [hunk 1]
```diff
@@ -114,83 +138,409 @@ impl CGDisplayMode {
...
+        unsafe { CGDisplayPixelsWide(self.id) as u64 }
+    }
+
+    /// Provides a list of displays that are active (or drawable).
+    #[inline]
+    pub fn active_displays() -> Result<Vec<CGDirectDisplayID>, CGError> {
+        let count = try!(CGDisplay::active_display_count());
+        let mut buf: Vec<CGDirectDisplayID> = vec![0; count];
+        let result = unsafe { CGGetActiveDisplayList(count as u32, buf.as_mut_ptr(), ptr::null_mut()) };
+        if result == 0 { Ok(buf) } else { Err(result) }
```
Comment (jdm) [inline src/display.rs]: More rustfmt going haywire and doing things twice.

File: src/display.rs [hunk 2]
```diff
@@ -114,83 +138,409 @@ impl CGDisplayMode {
...
+            Err(result)
+        }
+    }
+
+    /// Provides count of displays that are active (or drawable).
+    #[inline]
+    pub fn active_display_count() -> Result<usize, CGError> {
+        let mut count: libc::uint32_t = 0;
+        let result = unsafe { CGGetActiveDisplayList(0, ptr::null_mut(), &mut count) };
+        if result == 0 { Ok(count as usize) } else { Err(result) }
```
Comment (jdm) [inline src/display.rs]: We appear to be doing this twice.

File: src/display.rs [hunk 3]
```diff
@@ -10,13 +10,13 @@
 #![allow(non_upper_case_globals)]
 
 use libc;
-use std::mem;
```
Comment (jdm) [inline src/display.rs]: Please squash this commit into the previous two as appropriate.

Comment (msanders) [reply src/display.rs]: @jdm: Thanks, done.

**Pull request merged**